### PR TITLE
UCT/TCP: Calculate TX buffer offset when it is used

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Calculate TX buffer offset when it is used

## Why ?

This needs to be done after `sendv()` called - it updates `tx.offset` value

## How ?

Move under `uct_tcp_ep_ctx_buf_need_progress()` if statement

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12246/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 12-Sep-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/b4c6534850c4fdccb3152dea48316840f91513cf/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/b6c48ab87c16bffa16b923f85dec925e1902d724/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/a4f78fcb7163fc85d6388081643991516dff468c/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/e5c781ec11e16cff8b65972e60c654f3ada8456a/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/d1d485d6cf30a98551adf8f7ae400a6d70baa126/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/3b725b567dd77b7ed599a96ea74c8c1cae6f78e3/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/a7b0a844a11ca6b43ed604f66ba2a738b8881dc4/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/0a29b68d507b57e22a0d3952db7a540a4b1568d4/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/f3562cd44ca018376c1050a3451f01b7e54235bd/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/85c6bad51530b11ea1db103f48bf88f966da75db/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/75acb4376af84ad5835f3be40e8c45b740389736/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/dfd6e29b3dda0497a8a28fcc84402f33f8eeb0e9/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/598db77b4e21629d18bbd25de9f31dffdd71877b/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/b74b3ee7a98cba89e43b38b6b36c8f461481e4e6/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/4c9a637d11c8114bc322a79522659768308368b3/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/a7f90062f3a9aef1e89dc3ad60bbc1fb9959784c/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/99aa6f661dc5fe66bfebbf1b0da84562bd88f643/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/ca09f41608a71dfdb061ff42b962cb97c8853e5a/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/fc9684311c46d406529f7a40c75f3c7e3124e648/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/cce3cb7447e5b4cd91b30627c0fa61143ba3a312/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/7553224fbc2e3fe432aba1dc3a085994b26ca830/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/bba763e729eec75a48f8eaf1f8395041f183f333/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/82e2e9cc57f705c4c4599eb1d8a84eb8aa68caf2/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/09ba6ccc33da507492abfae066ffe2cb81a68251/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/e0ae4e16672d0f919072b69ccad20dac5bcddeb2/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/bd958604d5be82cc80da9a51c4b3ad52b26734bc/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/38fab0f57f1e6eb010bf0c55de940f107ed893de/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/b764c835b5af10cb05b31cf17d75e350fd2402be/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/9130863ecf8b369c113eac26fc129390/raw/a3d8c757405832323700fac9802a6cdacf6eb561/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by dmitrygx** [comment]
@brminich @hoopoepg could you review pls?

File: src/uct/tcp/tcp_ep.c [hunk 1]
```diff
@@ -928,9 +925,12 @@ ucs_status_t uct_tcp_ep_am_short(uct_ep_h uct_ep, uint8_t am_id, uint64_t header
             if (uct_tcp_ep_ctx_buf_need_progress(&ep->tx)) {
                 /* Copy only user's header and payload to the TX buffer,
                  * TCP AM header is placed at the beginning of the buffer */
+                offset = ((ep->tx.offset >= sizeof(*hdr)) ?
```
Comment (yosefe) [inline src/uct/tcp/tcp_ep.c]: maybe if `ep->tx.offset < sizeof(*hdr)`, we should not call ucs_iov_copy at all?

**Pull request merged**